crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -77,11 +77,25 @@ pub fn evaluate_main<'a>(
     igen: &mut IRGenerator<'a>,
     env: &mut Environment,
     main_func_body: HirFunction, //main function
+    constrain_main_return: bool,
 ) -> Result<(), RuntimeError> {
     let block = main_func_body.block(igen.def_interner());
-    for stmt_id in block.statements() {
-        igen.evaluate_statement(env, stmt_id)?;
+    let actual_return = igen.parse_block(block.statements(), env);
+
+    if constrain_main_return {
```

Comment:
I think it is better to do something like
```
if let Some(expected_return) = igen.find_variable(NodeInterner::main_return_id()) {
..
}
```
That way, you do not need to specify constrain_main_return , it implicitly comes from expected_return.
You should also merge from master for the constrain operation which is not binary anymore

Reply:
Done